include/vapor/Particle.h

```diff
@@ -0,0 +1,59 @@
+/* 
+ * Defines a particle used in flow integration.
+ */
+
+#ifndef PARTICLE_H
+#define PARTICLE_H
+
+#include <glm/glm.hpp>
+#include <forward_list>
+
+namespace flow
+{
+enum ERROR_CODE
+{
+    ADVECT_HAPPENED      =  1,
+    SUCCESS              =  0,
+    OUT_OF_FIELD         = -1,
+    NO_FIELD_YET         = -2,
+    NO_SEED_PARTICLE_YET = -3,
+    FILE_ERROR           = -4,
+    TIME_ERROR           = -5,
+    GRID_ERROR           = -6,
+    SIZE_MISMATCH        = -7
+};
+
+// Particle is not expected to serve as a base class.
+class Particle final
+{
+public:
+    glm::vec3   location { 0.0f, 0.0f, 0.0f };
+    float       time     = 0.0f;
+    float       value    = 0.0f;
+
+    // Constructors.
+    // This class complies with rule of zero.
+    Particle() = default;
+    Particle( const glm::vec3& loc, float t, float val = 0.0f );
+    Particle( const float* loc, float t, float val = 0.0f );
+    Particle( float x, float y, float z, float t, float val = 0.0f );
+
+    void  AttachProperty  (  float v );
```

Comment:
What is a "Property"?

Reply:
A property is a value that has a certain meaning associated with.

Reply:
Yes but what is it in this case? What is it used for?

Reply:
Oh, I see why you're confused. It's a data field that could be used to hold any information that the owner of particles want the particle object to hold. This data field is optional and could be anything that in addition to the fixed data fields such as time/position, etc.

Reply:
Yes I understand that but my question is what is it used for.

Reply:
Ok, the usage is better documented.